.github/workflows/build.yml

```diff
@@ -2,37 +2,95 @@ name: Build z390
 on: [push]
 
 jobs:
-  build:
+
+  build-distribution:
     runs-on: ubuntu-latest
+    outputs:
+      VERSION: ${{ steps.version.outputs.VERSION }}
     steps:
       - name: Checkout main
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: version
-        run: echo "version=$(git describe)" >> $GITHUB_OUTPUT
+        run: |
+          VERSION=$(git describe --tags)
+          echo "VERSION=${VERSION}" >> $GITHUB_OUTPUT
         id: version
-      - name: Build z390
-        uses: ./.github/actions/build
-      - name: release
-        uses: actions/create-release@v1
-        if: startsWith(github.ref, 'refs/tags/v')
+        shell: bash
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '8'
+      - name: Build z390 distribution
+        run: bash/blddist
+      - name: Save the distribution
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: ./dist/z390_${{ steps.version.outputs.VERSION }}.zip
+
+  run-tests:
+    runs-on: ubuntu-latest
+    needs: build-distribution
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '8'
+      - name: get dist artefact
+        uses: actions/download-artifact@v3
+        with:
+          name: dist
+      - name: Install dist and run tests
+        run: |
+          mkdir -p dist
+          unzip z390_${VERSION}.zip -d dist
+          ls -l dist/z390_${VERSION}
+          cp dist/z390_${VERSION}/z390.jar .
+          Z390_PROJECT_ROOT="../dist/z390_${VERSION}" z390test/gradlew -p z390test test 
```

Comment:
Question only. This looks like 2 statements: set Z390_PROJECT_ROOT and run z390test/gradlew. You can have 2 statements on the same line? I'm probably missing something.

Reply:
2 statements. One sets the envvar then runs the command. I believe the envvar scope is just for that command. I see it as an extension of the command.